### PR TITLE
Add 'latest' server options

### LINKED ISSUE #68: "runtime_config: server: tomcat" doesn't work
This configuration:

```yaml
runtime_config:
  server: tomcat
```
should select the Tomcat base image, but it currently gets rejected.
```
Step #0: Exception in thread "main" java.lang.IllegalArgumentException: Invalid jdk/server combination: *|tomcat
Step #0:        at com.google.cloud.runtimes.builder.config.domain.JdkServerLookup.lookupServerImage(JdkServerLookup.java:113)
Step #0:        at com.google.cloud.runtimes.builder.buildsteps.docker.DefaultDockerfileGenerator.generateDockerfile(DefaultDockerfileGenerator.java:57)
Step #0:        at com.google.cloud.runtimes.builder.buildsteps.docker.StageDockerArtifactBuildStep.doBuild(StageDockerArtifactBuildStep.java:92)
Step #0:        at com.google.cloud.runtimes.builder.buildsteps.base.BuildStep.run(BuildStep.java:52)
Step #0:        at com.google.cloud.runtimes.builder.BuildPipeline.build(BuildPipeline.java:62)
Step #0:        at com.google.cloud.runtimes.builder.Application.build(Application.java:81)
Step #0:        at com.google.cloud.runtimes.builder.Application.main(Application.jav
```

The readme does say that the server should be `tomcat8` or `jetty9`, but that seems to contradict the original design and simplicity in configuration.

_Edit: I originally thought it selected Jetty rather than Tomcat, but it actually just didn't deploy at all when selecting `server: tomcat`_.

### PR DESCRIPTION
**Pull request by aslo**
Adds "latest" server lookup names.

This PR allows users to select the server type without specifying a version, effectively using the :latest tag for each runtime. This enables app.yaml configurations that look like:

```yaml
# pins to tomcat:latest
runtime_config:
  server: tomcat
``` 
or 

```yaml
 # pins to tomcat:8
runtime_config:
  jdk: openjdk8
  server: tomcat
``` 

fixes #68

### CODE REVIEW

File: java.yaml [hunk 1]
```diff
@@ -11,9 +11,13 @@ steps:
   - '*|*=gcr.io/google-appengine/jetty:9'
   - 'openjdk8|*=gcr.io/google-appengine/jetty:9'
   - 'openjdk8|jetty9=gcr.io/google-appengine/jetty:9'
+  - 'openjdk8|jetty=gcr.io/google-appengine/jetty:9'
```
Comment (meltsufin) [inline java.yaml]: Why isn't it `gcr.io/google-appengine/jetty:latest`?

File: java.yaml [hunk 2]
```diff
@@ -11,9 +11,13 @@ steps:
...
   - 'openjdk8|tomcat8=gcr.io/google-appengine/tomcat:8'
+  - 'openjdk8|tomcat=gcr.io/google-appengine/tomcat:8'
```
Comment (meltsufin) [inline java.yaml]: Same here, why isn't it `:latest`?

File: java.yaml (hunk 1 above)
Comment (aslo) [inline java.yaml]: Presumably at some point in the future, the jetty:latest point to an image that's based on Java 9. When that's the case, it wouldn't make sense for the user to specify `jdk: openjdk8` and get a java 9 based runtime.

IMO the correct behavior is to select the latest tag of the jetty runtime that uses openjdk8 (which is jetty:9).

File: java.yaml (hunk 2 above)
Comment (aslo) [inline java.yaml]: Same reasoning as above.

File: java.yaml (hunk 1 above)
Comment (meltsufin) [inline java.yaml]: I see your reasoning now. If we never support more than one version of a server per version of JDK, it makes perfect sense to not use :latest when JDK is specified, but to use it when JDK is not specified.

**Pull request merged**